File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_csharp/src_wrapper/GnArtistEdit.cs

namespace GracenoteSDK {

/**
*  @internal GnArtistEdit @endinternal
*  represents  Artist's information returned by Gracenote.
*/
public class GnArtistEdit : GnDataObject {
  private global::System.Runtime.InteropServices.HandleRef swigCPtr;

  internal GnArtistEdit(global::System.IntPtr cPtr, bool cMemoryOwn) : base(gnsdk_csharp_marshalPINVOKE.GnArtistEdit_SWIGUpcast(cPtr), cMemoryOwn) {
    swigCPtr = new global::System.Runtime.InteropServices.HandleRef(this, cPtr);
  }

  internal static global::System.Runtime.InteropServices.HandleRef getCPtr(GnArtistEdit obj) {
    return (obj == null) ? new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero) : obj.swigCPtr;
  }

  ~GnArtistEdit() {
    Dispose();
  }

  public override void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != global::System.IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_GnArtistEdit(swigCPtr);
        }
        swigCPtr = new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero);
      }
      global::System.GC.SuppressFinalize(this);
      base.Dispose();
    }
  }

/**
*  @internal getNameOfficialEdit @endinternal
*  Create an empty object of GnNameEdit using NAME OFFICIAL.
*  @return GnNameEdit
*/
  public GnNameEdit Name {
    get {
      global::System.IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnArtistEdit_Name_get(swigCPtr);
      GnNameEdit ret = (cPtr == global::System.IntPtr.Zero) ? null : new GnNameEdit(cPtr, true);
      return ret;
    } 
  }

/**
*  @internal getContributorEdit @endinternal
*  Create an empty object of GnContributorEdit using CONTRIBUTOR.
*  @return GnContributorEdit
*/
  public GnContributorEdit Contributor {
    get {
      global::System.IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnArtistEdit_Contributor_get(swigCPtr);
      GnContributorEdit ret = (cPtr == global::System.IntPtr.Zero) ? null : new GnContributorEdit(cPtr, true);
      return ret;
    } 
  }

  public GnArtist GnArtist {
    get {
      global::System.IntPtr cPtr = gnsdk_csharp_marshalPINVOKE.GnArtistEdit_GnArtist_get(swigCPtr);
      GnArtist ret = (cPtr == global::System.IntPtr.Zero) ? null : new GnArtist(cPtr, true);
      return ret;
    } 
  }

}

}

File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/GnArtistEdit.java

package com.gracenote.gnsdk;

/** 
* 
*  represents  Artist's information returned by Gracenote. 
*/ 
 
public class GnArtistEdit extends GnDataObject {
  private transient long swigCPtr;

  protected GnArtistEdit(long cPtr, boolean cMemoryOwn) {
    super(gnsdk_javaJNI.GnArtistEdit_SWIGUpcast(cPtr), cMemoryOwn);
    swigCPtr = cPtr;
  }

  protected static long getCPtr(GnArtistEdit obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_GnArtistEdit(swigCPtr);
      }
      swigCPtr = 0;
    }
    super.delete();
  }

  public GnArtist gnArtist() {
    return new GnArtist(gnsdk_javaJNI.GnArtistEdit_gnArtist(swigCPtr, this), true);
  }

/** 
* 
*  Create an empty object of {@link GnNameEdit} using NAME OFFICIAL. 
*  @return {@link GnNameEdit} 
*/ 
 
  public GnNameEdit name() throws com.gracenote.gnsdk.GnException {
    return new GnNameEdit(gnsdk_javaJNI.GnArtistEdit_name(swigCPtr, this), true);
  }

/** 
* 
*  Create an empty object of {@link GnContributorEdit} using CONTRIBUTOR. 
*  @return {@link GnContributorEdit} 
*/ 
 
  public GnContributorEdit contributor() throws com.gracenote.gnsdk.GnException {
    return new GnContributorEdit(gnsdk_javaJNI.GnArtistEdit_contributor(swigCPtr, this), true);
  }

}
